internal/capability/auditor.go

```diff
@@ -26,6 +31,90 @@ type CapAuditor struct {
 
 	// AllInstallModes will test all install modes supported by an operator
 	AllInstallModes bool
+
+	// ExtraCustomResources associates packages to a list of Custom Resources (in addition to ALMExamples)
+	// to be audited by the OperandInstall AuditPlan.
+	ExtraCustomResources map[string]interface{}
+}
+
+// ExtraCRDirectory scans the provided directory and populates the ExtraCustomResources field.
+// Is is expected that the extraCRDirectory posesses subdirectories. Manifest files are present in each subdirectory.
+// The name of the subdirectory is used to determine which package the manifest files are corresponding to.
+// The resulting structure would be:
+// custom_resources_directory/
+// ├── package_name1
+// │   ├── manifest_file1.json
+// │   └── manifest_file2.yaml
+// └── package_name2
+//     ├── manifest_file1.json
+//     └── manifest_file2.yaml
+func (capAuditor *CapAuditor) ExtraCRDirectory(extraCRDirectory string) error {
+	logger.Debugw("scaning for extra Custom Resources", "extra CR directory", extraCRDirectory)
+	extraCustomResources := map[string]interface{}{} // maps packages to a list of CR
+
+	extraCRDirectoryAbsolutePath, err := filepath.Abs(extraCRDirectory)
+	if err != nil {
+		return fmt.Errorf("could not get absolute path from %s: %v", extraCRDirectory, err)
+	}
+
+	err = filepath.WalkDir(extraCRDirectoryAbsolutePath, func(path string, d fs.DirEntry, err error) error {
+		if err != nil && path == extraCRDirectoryAbsolutePath {
+			// Error reading the root directory, exit and return the error
+			return err
+		}
+
+		if !d.IsDir() { // Act on files only
+			manifestFilePath := path
+			// Checking that the manifest file is placed in a subdirectory
+			if len(strings.Split(manifestFilePath, "/")) != len(strings.Split(extraCRDirectoryAbsolutePath, "/"))+2 {
+				logger.Errorf("Error handling manifest file %s. File should be placed in a subdirectory of %s", manifestFilePath, extraCRDirectory)
+				return nil // continue
+			}
+
+			// Get the name of the subdirectory containing the manifest file. This corresponds to the package name.
+			packageName := filepath.Base(filepath.Dir(manifestFilePath))
+
+			logger.Debugw("adding Custom Resource", "source manifest file", manifestFilePath, "package", packageName)
+
+			// Get manifest file content
+			manifestFileContentRaw, err := os.ReadFile(manifestFilePath)
```

Comment:
Nit: Can we maybe shorten this var to manifestBytes or just manifestRaw?

Reply:
I don't have a strong opinion here.
I have a tendency for descriptive (end therefore sometimes long) names, but I guess that can be shorten.